core/controllers/suggestion.py

```diff
@@ -21,13 +21,61 @@
 from constants import constants
 from core.controllers import acl_decorators
 from core.controllers import base
+from core.domain import opportunity_services
 from core.domain import suggestion_services
 from core.platform import models
 import feconf
+import utils
 
 (suggestion_models,) = models.Registry.import_models([models.NAMES.suggestion])
 
 
+def _require_valid_suggestion_and_target_types(target_type, suggestion_type):
+    """Checks whether the given target_type and suggestion_type are valid.
+
+    Args:
+        target_type: str. The type of the suggestion target.
+        suggestion_type: str. The type of the suggestion.
+
+    Raises:
+        InvalidInputException: If the given target_type of suggestion_type are
+            invalid.
+    """
+    if target_type not in suggestion_models.TARGET_TYPE_CHOICES:
+        raise utils.InvalidInputException(
+            'Invalid target_type: %s' % target_type)
+
+    if suggestion_type not in suggestion_models.SUGGESTION_TYPE_CHOICES:
+        raise utils.InvalidInputException(
+            'Invalid suggestion_type: %s' % suggestion_type)
+
+
+def _get_exploration_suggestion_representable(suggestions):
```

Comment:
No passing dicts around. Maybe have this function just compute and return target_ids_to_opportunities, then compose the dict directly in the controller handler in each case.

Also this function should probably go in the domain layer (it's "logic").

Reply:
Done! (Removed this function as it wasn't required according to the new requirements.)